plugins/include/tse.inc

```diff
@@ -0,0 +1,266 @@
+
+// Returns player weapon entity (weapon_tsgun), if it not found returns 0
+native tse_getuserweapent(idx);
+
+// Sets player's weapon fire mode (TSE_FM_*), on error returns 0
+native tse_setweapfiremode(idx, weap, mode);
+
+// Gets player's weapon fire mode (TSE_FM_*), on error returns -1
+native tse_getweapfiremode(idx, weap);
+
+/* 
+* Sets player weapon attachments (TSE_ATM_*), on error returns 0
+* To sets multiple attacments, use their sum
+* For example: tse_setweapatcments(pl, TSE_WPN_M4A1, TSE_ATM_SILENCER + TSE_ATM_LASERSIGHT + TSE_ATM_SCOPE)
+*/
+native tse_setweapatcments(idx, weap, attachments, isactive=1);
+
+// Sets player weapon count of bullets in clip (for identify weapon use TSE_WPN_*), on error returns 0
+native tse_setweapclip(idx, weap, clip);
+
+// Sets player weapon count of bullets totally (for identify weapon use TSE_WPN_*), on error returns 0
+native tse_setweapammo(idx, weap, ammo);
+
+// Sets player weapon readiness status (works with pump-action weapons), on error returns 0
+native tse_setweapready(idx, weap, isready);
+
+// Checks if player has the weapon with the given ID, if it returns 1, otherwise 0
+native tse_isuserhasweap(idx, weap);
+
+// Creates powerup in specified position, with specified type and time of life (the time through which created powerup will disappear completely)
+native tse_createpwup(type, Float:coord[3], ttl=120);
+
+// Gives powerup to user with specified type and duration (in seconds), on error returns 0
+native tse_giveuserpwup(idx, type, duration);
+
+// Creates weapon in specified position, and with specified attachments (TSE_ATM_*), time of life, count of bullets in clip, on error returns 0
+native tse_createweap(weap, Float:coord[3], clip, atcments, ttl=120);
+
+// Gives weapon to user with specified count of bullets in clip and attachments (TSE_ATM_*), on error returns 0
+native tse_giveuserweap(idx, weap, clip, atcments=0);
+
+// Returns user current powerup (TSE_PWUP_*), on error returns 0
+native tse_getuserpwup(idx);
+
+// Returns user current state (TSE_STN_*), on error returns 0
+native tse_getuserstate(idx);
+
+// Returns user current status (text near nick in scoreboard) (TSE_STAT_*), on error returns -1
+native tse_getuserstatus(idx);
+
+// Sets user current status (TSE_STAT_*), on error returns 0
+native tse_setuserstatus(idx, status);
+
+// Returns user current active powerup (TSE_PWUP_*), on error returns -1
+native tse_getuseractivepwup(idx);
+
+// Sets user powerup duration (in seconds), on error returns 0
+native tse_setuserpwupduration(idx, duration);
+
+// Sets user fake powerup effect (as though user had a speficied powerup and user activated it), on error returns 0
+native tse_setfakepwup(idx, pwupid, Float:duration);
+
+/* Configures fake powerup options (player and effects physical speed, aura distance)
+*  WARNING: THIS FUNCTION WORKS ONLY WITH FUNCTION tse_setfakepwup AND POWERUP TYPE TSE_PWUP_SLOWMO OR TS_PWUP_SLOWPAUSE ONLY!
```

Comment:
You should emit native error if function is misused.

Reply:
Thanks for revision of my code, I will take action as soon as possible.

Reply:
Thanks for revision of my code, I will take action as soon as possible.